tck/src/main/java/org/eclipse/microprofile/health/tck/DelayedCheckTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICES file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+package org.eclipse.microprofile.health.tck;
+
+import org.eclipse.microprofile.health.tck.deployment.DelayedCheck;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.shrinkwrap.api.Archive;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+
+import static org.eclipse.microprofile.health.tck.DeploymentUtils.createWarFileWithClasses;
+
+/**
+ * @author Martin Stefanko
+ */
+public class DelayedCheckTest extends TCKBase {
+
+    @Deployment
+    public static Archive getDeployment() {
+        return createWarFileWithClasses(DelayedCheck.class);
+    }
+
+    /**
+     * Verifies the health integration with CDI when the bean creation takes longer than
+     * the first request
+     */
+    @Test
+    @RunAsClient
+    public void testSuccessResponsePayload() {
+        Response response = getUrlReadyContents();
+
+        // status code
+        Assert.assertEquals(response.getStatus(),503);
+
+        JsonObject json = readJson(response);
+
+        // response size
+        JsonArray checks = json.getJsonArray("checks");
+        Assert.assertEquals(checks.size(), 1, "Expected a single check response");
+
+        // single procedure response
+        assertFailureCheck(checks.getJsonObject(0), "delayed-check");
+
+        assertOverallFailure(json);
+    }
```

Comment:
Shouldn't we try again after the 3s of bean creation to make sure that the service is now ready (bean has been fully instantiated) ?

Reply:
The idea is that even this first request shouldn't return response until all checks (so the one that is delayed) is ready. Essentially it must block until all checks can be invoked. @antoinesd Is my assumption wrong?

Reply:
i guess it's ok then.